utils/gh_wrap_errors.py

```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/env python
+
+"""
+Utility script to run a command, and propagate potential GNU style errors as
+GitHub workflow commands.
+"""
+
+import argparse
+from os import path as P
+import re
+import select
+import subprocess
+import sys
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+_, extra_args = parser.parse_known_args()
+
+path = P.dirname(P.realpath(__file__))
+msg_re = re.compile(r"(.*?):(\d+):(?:(d+)\:)?(.*)?$")
+
+p = subprocess.Popen(
+    extra_args, universal_newlines=True,
+    stdout=subprocess.PIPE, stderr=subprocess.PIPE
+)
+
+exit_status = 0
+
+
+def emit_workflow_command_for_line(line):
+    """
+    If line is a GNU style error message, emit a corresponding workflow
+    command.
+    """
+    global exit_status
+    m = msg_re.match(line)
+    if m:
+        exit_status = 1
+        file, line, _, msg = m.groups()
+        if file.startswith('/'):
+            file = P.relpath(file)
+        print(f"::error file={file},line={line}::{msg}")
+
+
+stdout = []
+stderr = []
+
+# Record the subprocess's stdout & stderr while still feeding them to this
+# process' out & err.
```

Comment:
> Also, could you add this file to our mypy config? (stop me if I’m wrong, but I understood we agreed to add type hints for all new code)

TBH for me that clearly didn't cover scripts, that are really not part of langkit itself. That'd be like adding type hints to bash scripts, eg. the effort is not worth the benefit. Tell me if you disagree.

Reply:
> This does not correctly updates the exit status. In any case, IIUC this reimplements .communicate: could you replace all this business with:

Not from what I understand at least. This script will make sure that stderr and stdout are forwarded as they arrive, not in one shot at the end (or IOW this is more like `tee`). For looking at the output of those scripts while a CI is running, for debugging purposes for example, this is much more user friendly.

Reply:
Ok, I see. I think it would be worth adding comments to explain that. In my experience (admittedly from a long time ago), dealing with async IO with select is delicate, with problems similar to “on some conditions, the read can be blocking, so this can end up in a deadlock.

This script will be part of our dev/review workflow, so think we should stay as much away as possible from such subtle issues. Given that the stdout/stderr distinction does not really matter for us, I suggest to pass `stderr=subprocess.PIPE` to the `Popen` constructor above, and then simplify the reading loop so something like:

```
for line in p.stdout:
    print(line)
    emit_workflow_command_for_line(line)
```

As a bonus point (?) we get feedback to the workflow while the script is running (rather than all in once at the end of the script). What do you think?

Reply:
> What do you think?

I can try, I'm a bit wary about the supposedly innocuous merging of stdout and stderr, but I can't tell why, and you're probably right that we don't care.

Reply:
> As a bonus point (?) we get feedback to the workflow while the script is running (rather than all in once at the end of the script). What do you think?

This already works (see the sys.stdout.write in the loop)

Reply:
No, I meant: feedback with properly formatted messages. Right now if I understand correctly, they are emitted only once we have the full stdout+stderr content.